posts/Rust-1.86.0.md

```diff
@@ -0,0 +1,163 @@
+```rust
+let v = &mut [1, 2, 3];
+if let Ok([a, b]) = v.get_disjoint_mut([0, 2]) {
+    *a = 413;
+    *b = 612;
+}
+assert_eq!(v, &[413, 2, 612]);
+
+if let Ok([a, b]) = v.get_disjoint_mut([0..1, 1..3]) {
+    a[0] = 8;
+    b[0] = 88;
+    b[1] = 888;
+}
+assert_eq!(v, &[8, 88, 888]);
+
+if let Ok([a, b]) = v.get_disjoint_mut([1..=2, 0..=0]) {
+    a[0] = 11;
+    a[1] = 111;
+    b[0] = 1;
+}
+assert_eq!(v, &[1, 11, 111]);
+```
+
+### Allow safe functions to be marked with the #[target_feature] attribute.
+
+Functions marked with `#[target_feature]` are generally considered as unsafe functions: they are unsafe to call, cannot *generally* be assigned to safe function pointers, and don't implement the `Fn*` traits.
+
+However, calling them from other `#[target_feature]` functions with a superset of features is safe.
+
+```rust
+// Demonstration function
+#[target_feature(enable = "avx2")]
+fn avx2() {}
+
+fn foo() {
+    // Calling `avx2` here is unsafe, as we must ensure
+    // that AVX is available first.
+    unsafe {
+        avx2();
+    }
+}
+
+#[target_feature(enable = "avx2")]
+fn bar() {
+    // Calling `avx2` here is safe.
+    avx2();
+}
+```
+
+### Debug assert pointers are non-null when required for soundness
+
+The compiler will now insert checks that a pointer is not null upon non-zero-sized reads and writes, and also when the pointer is reborrowed into a reference. For example, the following code will now produce a non-unwinding panic:
+```rust
+let _x = *std::ptr::null::<u8>();
+let _x = &*std::ptr::null::<u8>();
+```
+Trivial examples like this have produced a warning since Rust 1.53.0, the new runtime check will detect these scenarios regardless of complexity.
+
+### Make `missing_abi` lint warn by default
+
+Omitting the ABI in extern blocks and functions (e.g. `extern {}` and `extern fn`) will now result in a warning (via the `missing_abi` lint). Omitting the ABI after the `extern` keyword has always implicitly resulted in the `"C"` ABI. It is now recommended to explicitly specify the `"C"` ABI (e.g. `extern "C" {}` and `extern "C" fn`).
+
+### Target deprecation warning for 1.87.0
+
+The target `i586-pc-windows-msvc` will be removed in the next version of Rust, 1.87.0. This is because this target is effectively useless. Its difference to the much more popular `i686-pc-windows-msvc` is that it does not require SSE2 instruction support, but Windows 10, the minimum required OS version of all `windows` targets (except the `win7` targets), requires SSE2 instructions itself, so this target does not fulfill a purpose.
```

Comment:
Would it make sense to make an explicit recommendation to migrate?

```suggestion
The target `i586-pc-windows-msvc` will be removed in the next version of Rust, 1.87.0. This is because this target is effectively useless. Its difference to the much more popular `i686-pc-windows-msvc` is that it does not require SSE2 instruction support, but Windows 10, the minimum required OS version of all `windows` targets (except the `win7` targets), requires SSE2 instructions itself, so this target does not fulfill a purpose.

Users currently targeting `i586-pc-windows-msvc` should migrate to `i686-pc-windows-msvc` before the 1.87.0 release.
```

Reply:
This migration is not always possible, maybe qualify it with "if SSE2 instructions are supported" then `i686-pc-windows-msvc` is a viable migration target. Otherwise, the user will likely have to build their own `i586-pc-windows-msvc`-esque toolchain.

Reply:
Added this, I feel like it reads a bit weird to qualify the migration and then not give an alternative but it also feels a bit weird to suggest people maintain their own toolchain lol

Reply:
The existing target already requires Windows 10 which requires SSE2. So migration should be always possible. IOW, the target made no logical sense. ;)

Reply:
That's what I had thought but the "except the win7 targets" makes it seem like that's an exception. But also we say here that windows 10 is the minimum required version for windows targets.... So I'm not really sure how we simultaneously can have win7 targets that need to be called out as an exception, but also we require atleast windows 10 :thinking:

Reply:
"Windows 10, the minimum required OS version of all `windows` targets (except the `win7` targets), requires SSE2 instructions itself" is a universal statement and correct. However it is confusing here because we only care about one target, `i586-pc-windows-msvc`, which is *not* a `win7` target.

Reply:
Oh I see, the win7 targets are named as `*-win7-windows.msvc`. I guess I got a bit mixed up between terms of art and the colloquial meaning of a "windows target" ^^